app/services/idv/session.rb

```diff
@@ -380,7 +380,7 @@ def new_idv_session
     end
 
     def session
-      user_session.fetch(:idv, {})
+      user_session[:idv] || {}
```

Comment:
As a further micro-optimization, I wonder if it could be useful to have a frozen empty hash constant reference somewhere we could use to further avoid allocations in these cases?

Like a...

```rb
EMPTY_HASH = {}.freeze
```

Poking around the Rails source, I did find a couple references to a constant used for this purpose, though I dunno that we'd want to use them directly:

- `ActiveRecord::QueryMethods::FROZEN_EMPTY_HASH`
- `ActionController::Parameters::EMPTY_HASH`

Reply:
Does the default get instantiated and persisted in the session? Like would we want something like this to preserve behavior?

```suggestion
      user_session[:idv] ||= {}
```

Reply:
> As a further micro-optimization, I wonder if it could be useful to have a frozen empty hash constant reference somewhere we could use to further avoid allocations in these cases?

I assumed it was likely that the hash is mutated, so I'm less sure if a frozen constant would be possible here.

> Does the default get instantiated and persisted in the session? Like would we want something like this to preserve behavior?

Not in the current implementation, the result of this is used in instantiation [here](https://github.com/18F/identity-idp/blob/aa462e3/app/services/idv/session.rb#L374-L376). I think it's in part due to the class being kind of unique in that it uses `method_missing` to allow for calling `idv_session.applicant = 'abc'` to set values in the hash. I'd like to refactor that since the mutation of the method [here](https://github.com/18F/identity-idp/blob/aa462e3/app/services/idv/session.rb#L101) also seems to be a bit of a hotspot, but that's a bigger effort.